From 82b702f9680471ba548866aa17756f76d980ee0a Mon Sep 17 00:00:00 2001
Date: Fri, 28 Aug 2020 09:13:33 +0800
Subject: [PATCH] Test SSLSocketSSLEngineTemplate.java intermittent failed with
 "Data length error"

Summary: <java.rmi>: Test SSLSocketSSLEngineTemplate.java intermittent failed with "Data length error"
LLT: jdk8u/jdk/test/sun/security/ssl/templates/SSLSocketSSLEngineTemplate.java
Bug url: 
---
 .../security/ssl/templates/SSLSocketSSLEngineTemplate.java    | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/jdk/test/sun/security/ssl/templates/SSLSocketSSLEngineTemplate.java b/jdk/test/sun/security/ssl/templates/SSLSocketSSLEngineTemplate.java
index 661941677..181bbc8fd 100644
--- a/jdk/test/sun/security/ssl/templates/SSLSocketSSLEngineTemplate.java
+++ b/jdk/test/sun/security/ssl/templates/SSLSocketSSLEngineTemplate.java
@@ -154,6 +154,8 @@ public class SSLSocketSSLEngineTemplate {
             SSLSocketSSLEngineTemplate test =
                 new SSLSocketSSLEngineTemplate(protocol);
             test.runTest(true);
+            // invalid session
+            test.invalidSession();
             test.runTest(false);
         }
 
@@ -488,4 +490,13 @@ public class SSLSocketSSLEngineTemplate {
             System.out.println(str);
         }
     }
+
+    /*
+     * invalid session
+     */
+    private void invalidSession() {
+        if (sslSocket != null && sslSocket.getSession() != null) {
+            sslSocket.getSession().invalidate();
+        }
+    }
 }
-- 
2.12.3

